### DISCUSSION TITLE
500 Error after upgrade 2.3.30

### DISCUSSION THREAD

**Discussion by andyrat33** [Security-Onion-Solutions/securityonion#3250] [Unsupported Versions]
This seems to have occurred after the recent upgrade but maybe there was an issue before that. so-status reports all green but the elasticsearch health reports an error. 
[root@so ~]# curl localhost:9200/_cluster/health/
{"cluster_name":"so","status":"green","timed_out":false,"number_of_nodes":1,"number_of_data_nodes":1,"active_primary_shards":831,"active_shards":831,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":0,"delayed_unassigned_shards":0,"number_of_pending_tasks":0,"number_of_in_flight_fetch":0,"task_max_waiting_in_queue_millis":0,"active_shards_percent_as_number":100.0}[root@so ~]# curl localhost:9200/_cat/health
{"error":{"root_cause":[{"type":"circuit_breaking_exception","reason":"[parent] Data too large, data for [<http_request>] would be [5036500752/4.6gb], which is larger than the limit of [5024566476/4.6gb], real usage: [5036500752/4.6gb], new bytes reserved: [0/0b], usages [request=0/0b, fielddata=7012/6.8kb, in_flight_requests=0/0b, accounting=714628484/681.5mb]","bytes_wanted":5036500752,"bytes_limit":5024566476,"durability":"PERMANENT"}],"type":"circuit_breaking_exception","reason":"[parent] Data too large, data for [<http_request>] would be [5036500752/4.6gb], which is larger than the limit of [5024566476/4.6gb], real usage: [5036500752/4.6gb], new bytes reserved: [0/0b], usages [request=0/0b, fielddata=7012/6.8kb, in_flight_requests=0/0b, accounting=714628484/681.5mb]","bytes_wanted":5036500752,"bytes_limit":5024566476,"durability":"PERMANENT"},"status":429}
It would be acceptable to clear the data if that would resolve the issue. 

[2021-03-02T23:59:48,556][WARN ][org.elasticsearch.monitor.jvm.JvmGcMonitorService] [gc][24299] overhead, spent [3.6s] collecting in the last [4.1s]
[2021-03-02T23:59:53,457][INFO ][org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService] attempting to trigger G1GC due to high heap usage [5223941736]
[2021-03-02T23:59:53,474][INFO ][org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService] GC did bring memory usage down, before [5223941736], after [4991747040], allocations [1], duration [17]
[2021-03-02T23:59:53,474][DEPRECATION][org.elasticsearch.deprecation.action.bulk.BulkRequestParser] [types removal] Specifying types in bulk requests is deprecated.
[2021-03-02T23:59:53,475][DEPRECATION][org.elasticsearch.deprecation.action.bulk.BulkRequestParser] [types removal] Specifying types in bulk requests is deprecated.
[2021-03-02T23:59:53,516][DEPRECATION][org.elasticsearch.deprecation.action.bulk.BulkRequestParser] [types removal] Specifying types in bulk requests is deprecated.
[2021-03-02T23:59:59,872][INFO ][org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService] attempting to trigger G1GC due to high heap usage [5163713504]
[2021-03-02T23:59:59,902][INFO ][org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService] GC did bring memory usage down, before [5163713504], after [5112339480], allocations [26], duration [30]

Clearing all the data from elasticsearch does not fix the problem.

Should I just reinstall?

**Comment by TOoSmOotH** [comment]
Looks like you need to increase the heap size of elastic.

**Reply by andyrat33** [reply]
Thanks, I don't know how to do that inside the container

**Reply by TOoSmOotH** [reply]
https://docs.securityonion.net/en/2.3/elasticsearch.html?highlight=heap#heap-size

**Comment by mbaki** [comment]
I'm having the same error, I have a standalone server with 16GB RAM (Small network), and I noticed when I click on alerts, I get a "Request failed with status code 500". In my elasticsearch log, I see tons of

[2021-05-01T17:22:56,358][INFO ][org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService] attempting to trigger G1GC due to high heap usage [4071223560]
[2021-05-01T17:22:56,390][INFO ][org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService] GC did bring memory usage down, before [4071223560], after [4067352904], allocations [1], duration [33]

I increased my elasticsearch heap size to esheap: '8092m' and still get the same status code.

[root@iaiso elasticsearch]# so-status 

Checking Docker status

    Docker ------------------------------------------------------------------------------------------------------------------ [ OK ]    

Checking container statuses

    so-aptcacherng ---------------------------------------------------------------------------------------------------- [ OK ]    
    so-cortex --------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-curator ------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-dockerregistry------------------------------------------------------------------------------------------------------ [ OK ]    
    so-elastalert ------------------------------------------------------------------------------------------------------- [ OK ]    
    so-elasticsearch ------------------------------------------------------------------------------------------------------- [ OK ]    
    so-filebeat --------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-fleet -------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-grafana ---------------------------------------------------------------------------------------------------------- [ OK ]    
    so-idstools -------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-influxdb --------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-kibana ------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-kratos -------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-logstash ----------------------------------------------------------------------------------------------------------- [ OK ]    
    so-mysql ----------------------------------------------------------------------------------------------------------- [ OK ]    
    so-nginx -------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-playbook --------------------------------------------------------------------------------------------------------- [ OK ]    
    so-redis ------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-sensoroni -------------------------------------------------------------------------------------------------------- [ OK ]    
    so-soc --------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-soctopus --------------------------------------------------------------------------------------------------------- [ OK ]    
    so-steno -------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-backend ---------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-coordinator ----------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-filestream ---------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-frontend -------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-gatekeeper --------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-manager ---------------------------------------------------------------------------------------------------- [ OK ]    
    so-suricata ------------------------------------------------------------------------------------------------------ [ OK ]    
    so-telegraf ------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-thehive ------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-thehive-es ------------------------------------------------------------------------------------------------------- [ OK ]    
    so-wazuh ----------------------------------------------------------------------------------------------------- [ OK ]    
    so-zeek ------------------------------------------------------------------------------------------------------------- [ OK ]    

 Thanks
Monah
